File: CIVITASRUBY/lib/ir_a_casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'casilla'
  class Ir_a_casilla < Sorpresa
    def initialize(tablero,valor,texto)
      super(tablero,valor,texto,nil)
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos) == true && @tablero != nil
        informe(actual,todos)
        casilla_actual = todos[actual].num_casilla_actual
        tirada = @tablero.calcular_tirada(casilla_actual, @valor)
        nueva = @tablero.nueva_posicion(casilla_actual, tirada)

        if todos[actual].mover_a_casilla(nueva) == true
          @tablero.get_casilla(nueva).recibe_jugador(actual, todos)
        end
      end
    end
    
  end
end

File: CIVITAS/src/civitas/IrACasilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class IrACasilla extends Sorpresa{
    IrACasilla(Tablero tablero , int valor , String texto){
        super(tablero ,valor, texto , null);
    }
    
    @Override
     void aplicarAJugador(int actual  ,ArrayList<Jugador> todos) {
        Tablero t = this.getTablero();
        if(this.jugadorCorrecto(actual, todos) && t != null){
            this.informe(actual, todos);
            int casillaActual = todos.get(actual).getNumCasillaActual();
            int tirada = t.calcularTirada(casillaActual, this.getValor());
            int nueva = t.nuevaPosicion(casillaActual, tirada);
            if(todos.get(actual).moverACasilla(nueva))
               t.getCasilla(nueva).recibeJugador(actual, todos);
        }
    
    }
}
